### DISCUSSION TITLE
Speaker speak 2 languages in an audio file

### DISCUSSION THREAD

**Discussion by toanhn** [SYSTRAN/faster-whisper#1041] [Q&A]
I have an audio file, and in it, the speaker speaks 2 languages, such as: English and Malay.
I transcribe the audio file 2 times, one time for English, and one time for Malay.
Can I base on the value "avg_logprob" to collect sentences for each language correctly?

**Comment by Hikari-Tsai** [comment]
No. `avg_logprob` measures how well the decoder supports the generated token sequence under the language it was asked to use; it is not a language classifier score. Comparing an English run with a Malay run can favor fluent but incorrect or hallucinated text and cannot reliably assign each sentence to a language.

For code-switched audio, use a multilingual model and enable per-window language detection:

```python
segments, info = model.transcribe(
    audio,
    multilingual=True,
    word_timestamps=True,
)
```

This lets the decoder change its language token for successive model windows. However, the current public `Segment` object does not expose a language field, and language changes are not guaranteed to align with sentence boundaries.

If you need an explicit language label per sentence, first obtain speech/sentence time ranges, run language identification on each range (for example with `model.detect_language(audio=clip)` or a dedicated spoken-language-ID model), and then transcribe that range using the selected `language`. Keep an “unknown/mixed” state for low-confidence clips; short utterances and accented speech are intrinsically difficult to classify.
